client/src/cmdlfnexwatch.c

```diff
@@ -257,12 +281,90 @@ static int CmdNexWatchReader(const char *Cmd) {
     }
 
     do {
-        lf_read(false, 20000);
+        lf_read(true, 20000);
         demodNexWatch(!cm);
     } while (cm && !kbd_enter_pressed());
     return PM3_SUCCESS;
 }
+//TOCHANGE
+
+static unsigned int bin2int(unsigned char *b, int size) {
+    unsigned int val = 0;
+    for (int i = 0; i < size; i++) {
+        if (b[i] == '1') {
+            val += pow(2, size - 1 - i);
+        }
+    }
+    return val;
+}
+static BYTE xor_parity(BYTE *stream_, int bit_number) {
+    BYTE parity_res = '0';
+    for (int i = bit_number - 1; i < 36; i += 4) {
+        if (stream_[i] == parity_res) {
+            parity_res = '0';
+        } else {
+            parity_res = '1';
+        }
+    }
+    return parity_res;
+}
 
+static BYTE *parity(BYTE *stream_) {
+    BYTE *parity_res = malloc(4 * sizeof(BYTE));
+    parity_res[0] = xor_parity(stream_, 4);
+    parity_res[1] = xor_parity(stream_, 2);
+    parity_res[2] = xor_parity(stream_, 3);
+    parity_res[3] = xor_parity(stream_, 1);
+    return parity_res;
+}
+
+static BYTE *convertUint8toByte(uint8_t number) {
+    BYTE *res = malloc(8 * sizeof(char));
+    uint8_t temp = number;
+    for (int i = 1; i < 9; i++) {
+        if (temp % 2) {
+            res[8 - i] = '1';
+        } else {
+            res[8 - i] = '0';
+        }
+        temp = temp / 2;
+    }
+    return res;
```

Comment:
mismatching fct name to what fct seem to be doing.   Ie  uint8 to binary string.